extension/lib/emailjs/emailjs-mime-parser.js

```diff
@@ -810,7 +810,10 @@
             }
 
             // override charset for text nodes
-            this.charset = this.contentType.params.charset = 'utf-8';
+            // issue #3520 - do not override ISO-2022-JP
+            if (this.charset?.toUpperCase() !== 'ISO-2022-JP') {
```

Comment:
I suggest to exclude based on `Content-Transfer-Encoding: 7bit`
As I mentioned before, 7bit is guaranteed to be not compatible with UTF-8.

Reply:
However, this header may be missing...
So we'd better check that either 7bit or it starts with (ISO-2022-JP...) (there is also ISO-2022-JP-2)

Reply:
@rrrooommmaaa How do I get value of `Content-Transfer-Encoding`?